tmt/result.py

```diff
@@ -317,49 +337,77 @@ def from_test_invocation(
             log=log or [],
             guest=ResultGuestData.from_test_invocation(invocation=invocation),
-            data_path=invocation.relative_test_data_path)
+            data_path=invocation.relative_test_data_path,
+            check=invocation.check_results or [])
 
-        return _result.interpret_result(invocation.test.result)
+        interpret_checks = {check.how: check.result for check in invocation.test.check}
 
-    def interpret_result(self, interpret: ResultInterpret) -> 'Result':
+        return _result.interpret_result(invocation.test.result, interpret_checks)
+
+    def interpret_result(
+            self,
+            interpret: ResultInterpret,
+            interpret_checks: dict[str, CheckResultInterpret]
+            ) -> 'Result':
         """
         Interpret result according to a given interpretation instruction.
 
         Inspect and possibly modify :py:attr:`result` and :py:attr:`note`
         attributes, following the ``interpret`` value.
 
         :param interpret: how to interpret current result.
+        :param interpret_checks: mapping of check:how and it's result interpret
         :returns: :py:class:`Result` instance containing the updated result.
         """
 
-        if interpret in (ResultInterpret.RESPECT, ResultInterpret.CUSTOM):
+        if interpret not in ResultInterpret:
+            raise tmt.utils.SpecificationError(
+                f"Invalid result '{interpret.value}' in test '{self.name}'."
+                )
+
+        if interpret == ResultInterpret.CUSTOM:
             return self
 
-        # Extend existing note or set a new one
-        if self.note:
-            self.note += f', original result: {self.result.value}'
+        original_result = self.result
+
+        # Interpret check results
+        self.check = [
+            check_result.interpret_check_result(CheckResultInterpret(interpret_checks[check_result.name]))
+            for check_result in self.check
+            ]
 
-        elif self.note is None:
-            self.note = f'original result: {self.result.value}'
+        if interpret not in (ResultInterpret.RESPECT, ResultInterpret.XFAIL):
+            self.result = ResultOutcome(interpret.value)
 
-        else:
-            raise tmt.utils.SpecificationError(
-                f"Test result note '{self.note}' must be a string.")
+            # Add original result to note if the result has changed
+            if self.result != original_result:
+                orig_note = f"original result: {original_result.value}"
+                self.note = f"{self.note}, {orig_note}" if self.note else orig_note
+
+            return self
+
+        failed_checks = [
+            check_result
+            for check_result in self.check
+            if check_result.result == ResultOutcome.FAIL
+            ]
+
+        if failed_checks:
+            self.result = ResultOutcome.FAIL
+            check_note = ", ".join([f"Check '{check.name}' failed" for check in failed_checks])
```

Comment:
```suggestion
            check_note = ", ".join([f"check '{check.name}' failed" for check in failed_checks])
```

A nitpick to make it look good when multiple checks fail. Now we get:

    Check 'avc' failed, Check 'dmesg' failed